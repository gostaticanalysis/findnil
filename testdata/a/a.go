package main

type T struct {
	N int
	m *int
}

var gt *T

func main() {
	f(new(T))
	f(g())
	println(gt.N)
	var t *T
	println(t.N)
	t2 := h(2)
	println(t2.N)
	var err error
	println(err.Error())
}

func f(t *T) {
	println(t.N)
}

func g() *T {
	return nil
}

func h(n int) *T {
	if n % 2 == 0 {
		return gt
	}
	return new(T)
}
