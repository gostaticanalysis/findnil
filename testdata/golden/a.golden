a/a.go:13:10 gt.N may be nil
a/a.go:15:10 t.N may be nil
a/a.go:17:10 t2.N may be nil
a/a.go:19:10 err.Error may be nil
a/a.go:23:10 t.N may be nil
